src/gluonts/dataset/arrow/dec.py

```diff
@@ -32,7 +32,7 @@ def from_schema(cls, schema):
         )
 
     def decode(self, batch, row_number: int):
-        yield from self.decode_batch(batch.slice(row_number, row_number + 1))
+        return next(self.decode_batch(batch.slice(row_number, row_number + 1)))
```

Comment:
Would this correctly return all elements in the batch?

Reply:
No, just one

Reply:
Okay, and is that the correct behavior?

Reply:
I think this is the intended behaviour, since the batch is being sliced to contain only one element; all elements in a batch (possibly containing a single element) can be generated via `decode_batch` directly, while this seems more of a conveniency method to directly access a particular element. A docstring would help, but this also seem to be the behaviour prior to #2217.

Reply:
`ArrowFile.__getitem__` is unchanged since #2217, so I think it got broken there and went unnoticed

Reply:
Ah okay. This is only slicing one element.

Reply:
Should we add a test for `__iter__` and `__getitem__` behaviors?

Reply:
Indeed: added assertion to existing test, failed before, passes now